bounce/server/api/membership.py

```diff
@@ -18,43 +18,75 @@ class MembershipEndpoint(Endpoint):
 
     __uri__ = "/memberships/<club_name:string>"
 
+<<<<<<< HEAD
+    @verify_token()
+    @validate(GetMembershipRequest, GetMembershipResponse)
+    async def get(self, request, club_name, id_from_token=None):
+=======
     @validate(GetMembershipsRequest, GetMembershipsResponse)
     async def get(self, request, club_name):
+>>>>>>> 66d04dfda62d525ba9715dc863eec8f2e201f0f9
         """
-        Handles a GET /memberships/<club_name>?user_id=<user_id> request
+        Handles a GET /memberships/<club_name> request
         by returning the membership that associates the given user with the
         given club. If no user ID is given, returns all memberships for the
         given club.
         """
         # Decode the club name
         club_name = unquote(club_name)
-        user_id = request.args.get('user_id', None)
 
         # Make sure the club exists
         club_row = club.select(self.server.db_session, club_name)
         if not club_row:
             raise APIError('No such club', status=404)
 
+<<<<<<< HEAD
+        body = util.strip_whitespace(request.json)
+        user_id = body.get('user_id', None)
+
+        try:
+            membership_attr = membership.select(self.server.db_session,
+                                                club_name, id_from_token,
+                                                Roles.president)
```

Comment:
Since anyone can select we should probably use Role.member just to be safe.

Reply:
Right, thanks for pointing that out, I'll change that.